stats.go

```diff
@@ -0,0 +1,1306 @@
+	Timestamp() StatsTimestamp
+	// Type returns the object's StatsType
+	Type() StatsType
+	// ID returns a unique id that is associated with the component inspected to produce
+	// this Stats object. Two Stats objects will have the same ID if they were produced
+	// by inspecting the same underlying object.
+	ID() string
+}
+
+// StatsType indicates the type of the object that a Stats object represents.
+type StatsType string
+
+const (
+	// StatsTypeCodec is used by CodecStats.
+	StatsTypeCodec StatsType = "codec"
+	// StatsTypeInboundRTP is used by InboundRTPStreamStats.
+	StatsTypeInboundRTP StatsType = "inbound-rtp"
+	// StatsTypeOutboundRTP is used by OutboundRTPStreamStats.
+	StatsTypeOutboundRTP StatsType = "outbound-rtp"
+	// StatsTypeRemoteInboundRTP is used by RemoteInboundRTPStreamStats.
+	StatsTypeRemoteInboundRTP StatsType = "remote-inbound-rtp"
+	// StatsTypeRemoteOutboundRTP is used by RemoteOutboundRTPStreamStats.
+	StatsTypeRemoteOutboundRTP StatsType = "remote-outbound-rtp"
+	// StatsTypeCSRC is used by RTPContributingSourceStats.
+	StatsTypeCSRC StatsType = "csrc"
+	// StatsTypePeerConnection used by PeerConnectionStats.
+	StatsTypePeerConnection StatsType = "peer-connection"
+	// StatsTypeDataChannel is used by DataChannelStats.
+	StatsTypeDataChannel StatsType = "data-channel"
+	// StatsTypeStream is used by MediaStreamStats.
+	StatsTypeStream StatsType = "stream"
+	// StatsTypeTrack is used by SenderVideoTrackAttachmentStats and SenderAudioTrackAttachmentStats.
+	StatsTypeTrack StatsType = "track"
+	// StatsTypeSender is used by by the AudioSenderStats or VideoSenderStats depending on kind.
+	StatsTypeSender StatsType = "sender"
+	// StatsTypeReceiver is used by the AudioReceiverStats or VideoReceiverStats depending on kind.
+	StatsTypeReceiver StatsType = "receiver"
+	// StatsTypeTransport is used by TransportStats.
+	StatsTypeTransport StatsType = "transport"
+	// StatsTypeCandidatePair is used by ICECandidatePairStats.
+	StatsTypeCandidatePair StatsType = "candidate-pair"
+	// StatsTypeLocalCandidate is used by ICECandidateStats for the local candidate.
+	StatsTypeLocalCandidate StatsType = "local-candidate"
+	// StatsTypeRemoteCandidate is used by ICECandidateStats for the remote candidate.
+	StatsTypeRemoteCandidate StatsType = "remote-candidate"
+	// StatsTypeCertificate is used by CertificateStats.
+	StatsTypeCertificate StatsType = "certificate"
+)
+
+// StatsTimestamp is a timestamp represented by the floating point number of
+// milliseconds since the epoch.
+type StatsTimestamp float64
+
+// Time returns the time.Time represented by this timestamp.
+func (s StatsTimestamp) Time() time.Time {
+	millis := float64(s)
+	nanos := int64(millis * float64(time.Millisecond))
+
+	return time.Unix(0, nanos).UTC()
+}
+
+// StatsReport collects Stats objects indexed by their ID.
+type StatsReport map[string]Stats
+
+// StatsBase contains all the fields Stats objects share
```

Comment:
Does this need to be public?

Reply:
On second thought, let's just delete it.

If I get rid of the `Stats` methods like I mention below all of these fields can just be duplicated in the base stat objects.